libcxx/include/__stacktrace/basic_stacktrace.h

```diff
@@ -0,0 +1,342 @@
+    size_type __skip      = 0;
+    size_type __max_depth = __default_max_depth;
+    _LIBCPP_ASSERT_VALID_ELEMENT_ACCESS(
+        __skip <= __skip + __max_depth, "sum of skip and max_depth overflows size_type");
+    basic_stacktrace __ret{__caller_alloc};
+    __ret.current_impl(__skip, __max_depth);
+    return __ret;
+  }
+
+  _LIBCPP_ALWAYS_INLINE static basic_stacktrace
+  current(size_type __skip, const allocator_type& __caller_alloc = allocator_type()) noexcept(__kNoThrowAlloc) {
+    size_type __max_depth = __default_max_depth;
+    _LIBCPP_ASSERT_VALID_ELEMENT_ACCESS(
+        __skip <= __skip + __max_depth, "sum of skip and max_depth overflows size_type");
+    basic_stacktrace __ret{__caller_alloc};
+    __ret.current_impl(__skip, __max_depth);
+    return __ret;
+  }
+
+  _LIBCPP_ALWAYS_INLINE static basic_stacktrace
+  current(size_type __skip,
+          size_type __max_depth,
+          const allocator_type& __caller_alloc = allocator_type()) noexcept(__kNoThrowAlloc) {
+    _LIBCPP_ASSERT_VALID_ELEMENT_ACCESS(
+        __skip <= __skip + __max_depth, "sum of skip and max_depth overflows size_type");
+    basic_stacktrace __ret{__caller_alloc};
+    if (__max_depth) [[likely]] {
+      __ret.current_impl(__skip, __max_depth);
+    }
+    return __ret;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI constexpr ~basic_stacktrace() = default;
+
+  static_assert(sizeof(__stacktrace::entry_base) == sizeof(stacktrace_entry));
+
+  _LIBCPP_HIDE_FROM_ABI explicit basic_stacktrace(const allocator_type& __alloc)
+      : base(__alloc, entry_iters_fn(), entry_append_fn()), __alloc_(__alloc), __entries_(__alloc_) {}
+
+  _LIBCPP_HIDE_FROM_ABI basic_stacktrace(basic_stacktrace const& __other, allocator_type const& __alloc)
+      : base(__alloc, entry_iters_fn(), entry_append_fn()), __alloc_(__alloc), __entries_(__other.__entries_) {}
+
+  _LIBCPP_HIDE_FROM_ABI basic_stacktrace(basic_stacktrace&& __other, allocator_type const& __alloc)
+      : base(__alloc, entry_iters_fn(), entry_append_fn()),
+        __alloc_(__alloc),
+        __entries_(std::move(__other.__entries_)) {}
+
+  _LIBCPP_HIDE_FROM_ABI basic_stacktrace() noexcept(is_nothrow_default_constructible_v<allocator_type>)
+      : basic_stacktrace(allocator_type()) {}
+
+  _LIBCPP_HIDE_FROM_ABI basic_stacktrace(basic_stacktrace const& __other) noexcept
+      : basic_stacktrace(__other, _ATraits::select_on_container_copy_construction(__other.__alloc_)) {}
+
+  _LIBCPP_HIDE_FROM_ABI basic_stacktrace(basic_stacktrace&& __other) noexcept
+      : basic_stacktrace(std::move(__other), __other.__alloc_) {}
+
+  _LIBCPP_HIDE_FROM_ABI basic_stacktrace& operator=(const basic_stacktrace& __other) {
+    if (std::addressof(__other) != this) {
+      if (__kPropOnCopyAssign) {
+        new (this) basic_stacktrace(__other, __other.__alloc_);
+      } else {
+        new (this) basic_stacktrace(__other);
+      }
+    }
+    return *this;
```

Comment:
Why do you construct a new `basic_stacktrace`? This leaks memory. Please add a regression test.